functions/PayrollProcessor.Functions/Features/Employees/Employee.cs

```diff
@@ -17,9 +19,41 @@ public class Employee
         public IEnumerable<EmployeePayroll> Payrolls { get; set; } = new EmployeePayroll[] { };
         public string Version { get; set; } = "";
 
-        public Employee(Guid id)
+        public Employee(Guid id) => Id = id;
+
+        public void UpdatePayrolls(Payroll payroll)
         {
-            Id = id;
+            var employeePayroll = new EmployeePayroll
+            {
+                Id = payroll.Id,
+                CheckDate = payroll.CheckDate,
+                GrossPayroll = payroll.GrossPayroll,
+                PayrollPeriod = payroll.PayrollPeriod
+            };
+
+            if (Payrolls.Length() < 30)
+            {
+                Payrolls = Payrolls.Prepend(employeePayroll);
+
+                return;
+            }
+
+            /*
+             * Find and (if found) replace the least older of all payrolls
+             * older than the one provided
+             * This keeps the set attached to the Employee as the latest 30 by CheckDate
+             */
+            var leastOlderPayroll = Payrolls
```

Comment:
I just realized I could do this same thing by adding the `payroll` to the `Payrolls` collection, sorting by date desc and then taking the first 30... 🤔

Reply:
Yeah prepending is an interesting solution to this query.